src/workflows-app/components/SubmitWorkflowModal.ts

```diff
@@ -0,0 +1,229 @@
+  method: WorkflowMethod;
+  methodVersion: MethodVersion;
+  recordType: string;
+  selectedRecords: object;
+  inputDefinition: InputDefinition[];
+  outputDefinition: OutputDefinition[];
+  callCachingEnabled: boolean;
+  onDismiss: () => any;
+  name: string;
+  namespace: string;
+  workspace: WorkspaceWrapper;
+};
+
+export const SubmitWorkflowModal = ({
+  method,
+  methodVersion,
+  recordType,
+  selectedRecords,
+  inputDefinition,
+  outputDefinition,
+  callCachingEnabled,
+  onDismiss,
+  name,
+  namespace,
+  workspace,
+  workspace: {
+    workspace: { workspaceId, createdBy },
+    canCompute,
+  },
+}: SubmitWorkflowModalProps) => {
+  const [runSetName, setRunSetName] = useState(
+    `${_.kebabCase(method.name)}_${_.kebabCase(recordType)}_${new Date().toISOString().slice(0, -5)}`
+  );
+  const [runSetDescription, setRunSetDescription] = useState('');
+  const [isCromwellRunnerLaunching, setIsCromwellRunnerLaunching] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [workflowSubmissionError, setWorkflowSubmissionError] = useState<string>();
+
+  const { captureEvent } = useMetricsEvent();
+  const canSubmit =
+    (isFeaturePreviewEnabled(ENABLE_AZURE_COLLABORATIVE_WORKFLOW_RUNNERS) && canCompute) ||
+    getTerraUser().email === createdBy;
+
+  const submitRun = async () => {
+    const runSetsPayload = {
+      run_set_name: runSetName,
+      run_set_description: runSetDescription,
+      method_version_id: methodVersion.method_version_id,
+      workflow_input_definitions: _.map(convertArrayType, inputDefinition),
+      workflow_output_definitions: outputDefinition,
+      wds_records: {
+        record_type: recordType,
+        record_ids: _.keys(selectedRecords),
+      },
+      call_caching_enabled: callCachingEnabled,
+    };
+    const {
+      cbasProxyUrlState: { state: cbasUrl },
+    } = await loadAppUrls(workspaceId, 'cbasProxyUrlState');
+    const runSetObject = await Ajax().Cbas.runSets.post(cbasUrl, runSetsPayload);
+    notify('success', 'Workflow successfully submitted', {
+      message: 'You may check on the progress of workflow on this page anytime.',
+      timeout: 5000,
+    });
+    captureEvent(Events.workflowsAppLaunchWorkflow, {
+      ...extractWorkspaceDetails(workspace),
+      methodUrl: methodVersion.url,
+      methodVersion: methodVersion.name,
+      methodSource: method.source,
+      previouslyRun: method.last_run.previously_run,
+    });
+    Nav.goToPath('workspace-workflows-app-submission-details', {
+      name,
+      namespace,
+      submissionId: runSetObject.run_set_id,
+    });
+  };
+
+  const submitToWorkflowsApp = () =>
+    poll(async () => {
+      try {
+        const appsResponse = await Ajax().Apps.listAppsV2(workspaceId, { role: 'creator' });
+        const appToUse =
+          getCurrentApp(appToolLabels.CROMWELL, appsResponse) ??
+          getCurrentApp(appToolLabels.CROMWELL_RUNNER_APP, appsResponse);
```

Comment:
We could, but non-creator users will be able to view this page if the workflows app is running, and will only be able to hit submit if the feature flag is enabled. This also holds consistent with the creator experience, since creators wouldn't need to have the manual launch feature flag enabled to launch a cromwell runner.